source/slang/slang-emit.cpp

```diff
@@ -1083,6 +1089,30 @@ Result linkAndOptimizeIR(
     if (requiredLoweringPassSet.optionalType)
         lowerOptionalType(irModule, sink);
 
+    if (requiredLoweringPassSet.nonVectorCompositeSelect)
+    {
+        switch (target)
+        {
+        case CodeGenTarget::HLSL:
+            legalizeNonVectorCompositeSelect(targetRequest, irModule, sink);
+            break;
+
+        case CodeGenTarget::SPIRV:
+        case CodeGenTarget::SPIRVAssembly:
+        {
+            // SPIRV older than 1.4 must legalize composites
+            // returned from a `select` instruction
+            if (codeGenContext->getTargetProgram()->getOptionSet().getProfileVersion() <= ProfileVersion::SPIRV_1_3)
```

Comment:
This should be deferred to spirv-legalize pass because we may upgrade that profile based on actual intrinsic usage. In spirv-legalize, we will first determine spirv version, and then run necessary steps based on knowledge of spirv version.